index.ts

```diff
@@ -25,12 +27,16 @@ app.get(
       level: Number(req.params.level),
     });
 
-    // TODO make this cached somehow so it doesn't get refetched every single time
-    // however, ensure that after sometime (maybe every hour or so) it _will_ fetch
-    // this fresh
-    const courses = await getCoursesFromSemester(coursesFromSemesterParams);
-
-    res.json(courses);
+    const id = req.originalUrl;
+    if (appCache.has(id)) {
+      console.log("Fetching data from cache...");
```

Comment:
convert these console logs to calls to https://www.npmjs.com/package/debug so they can be disabled in production but kept for local dev